tests/spicy/types/sink/parse-error.spicy

```diff
@@ -0,0 +1,29 @@
+# @TEST-EXEC:  printf '1' | spicy-driver -d %INPUT >>output 2>&1
```

Comment:
Could you add a test for `skip_delivery`, e.g., a test where we check that once a unit behind a sink enters an error state feeding it more data does not trigger additional errors (e.g., `more data after sink's unit has already completed parsing`)?

Reply:
Ok, extended the test case. 

Also added one more check for `skip_delivery` at closing time. Not sure how to trigger that, but seems it should be there.